### PR TITLE
Fix Readthedocs build failures

### PR DESCRIPTION
**Pull request by rrsettgast**
ever since the merge of #1184 the RTD build of develop fails with:
```
lookup cache used 12296/65536 hits=138610 misses=13248
finished...
loading translations [en]... done

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/develop/lib/python3.7/site-packages/sphinx/registry.py", line 472, in load_extension
    mod = __import__(extname, None, None, ['setup'])
ModuleNotFoundError: No module named 'matplotlib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/develop/lib/python3.7/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/develop/lib/python3.7/site-packages/sphinx/application.py", line 228, in __init__
    self.setup_extension(extension)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/develop/lib/python3.7/site-packages/sphinx/application.py", line 449, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/develop/lib/python3.7/site-packages/sphinx/registry.py", line 475, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname, err)
sphinx.errors.ExtensionError: Could not import extension matplotlib.sphinxext.only_directives (exception: No module named 'matplotlib')

Extension error:
Could not import extension matplotlib.sphinxext.only_directives (exception: No module named 'matplotlib')
```

### CODE REVIEW

File: src/conf.py [hunk 1]
```diff
@@ -100,10 +100,10 @@
...
     'sphinx.ext.imgmath',
     'sphinx.ext.todo',
-    'matplotlib.sphinxext.only_directives',
-    'matplotlib.sphinxext.plot_directive',
-    'IPython.sphinxext.ipython_directive',
-    'IPython.sphinxext.ipython_console_highlighting',
+#    'matplotlib.sphinxext.only_directives',
+#    'matplotlib.sphinxext.plot_directive',
+#    'IPython.sphinxext.ipython_directive',
+#    'IPython.sphinxext.ipython_console_highlighting',
```
Comment (rrsettgast) [inline src/conf.py:106]: @francoishamon Do you know why these were added?

Comment (castelletto1) [reply src/conf.py:106]: They were added to enable plotting capabilities: see [here](https://github.com/GEOSX/GEOSX/blob/develop/src/docs/sphinx/tutorials/step08/Tutorial.rst#inspecting-results). In other words, we can include python scripts directly in the documentation.

**Comment by francoishamon** [comment]
We modified the `conf.py` file to be able to plot the result of the time history output, for example in the _Visualization of results_ section of this [tutorial](https://github.com/GEOSX/GEOSX/blob/develop/src/docs/sphinx/tutorials/step05/Tutorial.rst). We wrote some python code to process the time history output directly in the .rst file to (1) display a png image in the tutorial and (2) allow the reader of the tutorial to click on `Source Code` and access the underlying python code.

It works fine with `make geosx_docs` on Quartz (no build error) and locally on Ubuntu with matplotlib installed, but does not seem to display correctly the images in the online documentation. As far as I can see on the `bugfix/rtdFailure` branch of the doc the images are still not displayed correctly. I can try to see what is going on.

**Comment by rrsettgast** [comment]
> We modified the `conf.py` file to be able to plot the result of the time history output, for example in the _Visualization of results_ section of this [tutorial](https://github.com/GEOSX/GEOSX/blob/develop/src/docs/sphinx/tutorials/step05/Tutorial.rst). We wrote some python code to process the time history output directly in the .rst file to (1) display a png image in the tutorial and (2) allow the reader of the tutorial to click on `Source Code` and access the underlying python code.
> 
> It works fine with `make geosx_docs` on Quartz (no build error) and locally on Ubuntu with matplotlib installed, but does not seem to display correctly the images in the online documentation. As far as I can see on the `bugfix/rtdFailure` branch of the doc the images are still not displayed correctly. I can try to see what is going on.

They don't display because I disabled the extensions in order to get the build working. We have to figure out how to re-enable the extensions.

**Comment by castelletto1** [comment]
Here is where I got those options: https://matplotlib.org/sampledoc/extensions.html

**Comment by francoishamon** [comment]
Could it be fixed by adding `autodoc_mock_imports` to our `conf.py` because the `matplotlib` library depends on C? See [this](https://docs.readthedocs.io/en/stable/faq.html#i-get-import-errors-on-libraries-that-depend-on-c-modules) in the RTD FAQ. If this might be the problem, can I try it on this branch and then revert if it does not work?

**Comment by klevzoff** [comment]
Our RTD config does not specify a pip requirements file, as such additional modules such as `matplotlib` are simply not installed for the build. The fix should be as simple as adding one https://docs.readthedocs.io/en/stable/config-file/v2.html#requirements-file

**Comment by rrsettgast** [comment]
The incision of a `requirements.txt` file moves the problem along:
```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/bugfix-rtdfailure/lib/python3.7/site-packages/sphinx/registry.py", line 472, in load_extension
    mod = __import__(extname, None, None, ['setup'])
ModuleNotFoundError: No module named 'matplotlib.sphinxext.only_directives'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/bugfix-rtdfailure/lib/python3.7/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/bugfix-rtdfailure/lib/python3.7/site-packages/sphinx/application.py", line 228, in __init__
    self.setup_extension(extension)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/bugfix-rtdfailure/lib/python3.7/site-packages/sphinx/application.py", line 449, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/envs/bugfix-rtdfailure/lib/python3.7/site-packages/sphinx/registry.py", line 475, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname, err)
sphinx.errors.ExtensionError: Could not import extension matplotlib.sphinxext.only_directives (exception: No module named 'matplotlib.sphinxext.only_directives')

Extension error:
Could not import extension matplotlib.sphinxext.only_directives (exception: No module named 'matplotlib.sphinxext.only_directives')
```

@francoishamon Where are the sphinext files for matplotlib?

**Comment by francoishamon** [comment]
@rrsettgast @castelletto1 Now the RTD build passes without error on this branch:
https://readthedocs.com/projects/geosx-geosx/builds/470061/

I added some requirements (`numpy`, `h5py`, `mpmath`) and removed some unnecessary extensions from `conf.py` (`matplotlib.sphinxext.only_directives`, `IPython`). I also fixed some unrelated warnings in `Outputs.rst` and `TasksManager.rst`.

In the online documentation, the problematic tutorials now correctly display the images:
https://geosx-geosx.readthedocs-hosted.com/en/bugfix-rtdfailure/docs/sphinx/tutorials/step05/Tutorial.html#visualization-of-results
https://geosx-geosx.readthedocs-hosted.com/en/bugfix-rtdfailure/docs/sphinx/tutorials/step08/Tutorial.html#inspecting-results

**Review by castelletto1 — APPROVED** [review]
This looks good to me!

**Pull request merged**